tests/steps/application_v2_steps.py

```diff
@@ -1157,13 +1173,79 @@ def check_box_contents(
 
 
 @then(
-    'the current application should have the following boxes "{box_names:MaybeString}".'
+    'according to "{from_client}", by parameter max {limit}, the current application should have {expected_num} boxes.'
+)
+def check_box_num_by_limit(context, from_client, limit, expected_num: str):
+    limit_int = int(limit)
+    expected_num_int = int(expected_num)
+    if from_client == "algod":
+        box_response = context.app_acl.application_boxes(
+            context.current_application_id, limit=limit_int
+        )
+    elif from_client == "indexer":
+        box_response = context.app_icl.application_boxes(
+            context.current_application_id, limit=limit_int
+        )
+    else:
+        assert False, "expecting algod or indexer, get " + from_client
+
+    assert expected_num_int == len(box_response["boxes"]), "expected box num"
+
+
+@then(
+    'according to "{from_client}", the current application should have the following boxes "{box_names:MaybeString}".'
 )
-def check_all_boxes(context, box_names: str = None):
-    expected_box_names = split_and_process_app_args(box_names)
-    box_response = context.app_acl.application_boxes(
-        context.current_application_id
+def check_all_boxes(context, from_client: str, box_names: str = None):
+    expected_box_names = []
+    if box_names:
+        expected_box_names = [
+            base64.b64decode(box_encoded)
+            for box_encoded in box_names.split(":")
+        ]
+    if from_client == "algod":
+        box_response = context.app_acl.application_boxes(
+            context.current_application_id
+        )
+    elif from_client == "indexer":
+        box_response = context.app_icl.application_boxes(
+            context.current_application_id
+        )
+    else:
+        assert False, "expecting algod or indexer, get " + from_client
+
+    actual_box_names = []
+    for box in box_response["boxes"]:
+        box = box["name"]
+        decoded_box = base64.b64decode(box)
+        actual_box_names.append(decoded_box)
+
+    # Check that length of lists are equal, then check for set equality.
+    assert len(expected_box_names) == len(
+        actual_box_names
+    ), f"Expected box names array length does not match actual array length {(expected_box_names)} != {(actual_box_names)}"
+    assert set(expected_box_names) == set(
+        actual_box_names
+    ), f"Expected box names array does not match actual array {expected_box_names} != {actual_box_names}"
+
+
+@then(
+    'according to indexer, by parameter max {limit} and next "{next_page:MaybeString}", the current application should have the following boxes "{box_names:MaybeString}".'
+)
+def check_all_boxes_by_indexer(
```

Comment:
Is this step made obsolete by `check_all_boxes()` since it handles `indexer` as well?

Reply:
no it's not because of the `next` parameter.

Resolving